crates/core/src/protocols/sumcheck/v3/bivariate_mlecheck.rs

```diff
@@ -0,0 +1,578 @@
+// Copyright 2025 Irreducible Inc.
+
+use std::{iter, mem, slice};
+
+use binius_compute::{
+	ComputeLayer, ComputeMemory, FSlice, FSliceMut, KernelBuffer, KernelMemMap, SizedSlice,
+	SlicesBatch,
+	alloc::{BumpAllocator, ComputeAllocator, HostBumpAllocator},
+};
+use binius_field::{
+	Field, TowerField,
+	util::{eq, powers},
+};
+use binius_math::{ArithExpr, CompositionPoly, EvaluationOrder, evaluate_univariate};
+use binius_utils::bail;
+
+use super::bivariate_product::{PhaseState, SumcheckMultilinear};
+use crate::{
+	composition::{BivariateProduct, IndexComposition},
+	protocols::sumcheck::{
+		CompositeSumClaim, EqIndSumcheckClaim, Error, RoundCoeffs, prove::SumcheckProver,
+	},
+};
+
+pub struct BivariateMLEcheckProver<'a, 'alloc, F: Field, Hal: ComputeLayer<F>> {
```

Comment:
Public struct needs docstring. You can use AI to generate and then read/edit yourself.